src/core/thread/mle.cpp

```diff
@@ -1416,6 +1416,25 @@ otError Mle::AppendAddressRegistration(Message &aMessage, AddressRegistrationMod
     VerifyOrExit(aMode != kAppendMeshLocalOnly, OT_NOOP);
     counter++;
 
+#if OPENTHREAD_CONFIG_TMF_DUA_REGISTRATION_ENABLE
+    if (Get<BackboneRouter::Leader>().HasDomainPrefix())
+    {
+        const Ip6::Address &address = GetDomainUnicastAddress();
+
+        if (Get<ThreadNetif>().IsUnicastAddress(address))
+        {
+            OT_ASSERT(Get<NetworkData::Leader>().GetContext(address, context) == OT_ERROR_NONE);
+
+            // Prioritize DUA, compressed entry
+            entry.SetContextId(context.mContextId);
+            entry.SetIid(address.GetIid());
+            SuccessOrExit(error = aMessage.Append(&entry, entry.GetLength()));
+            length += entry.GetLength();
+            counter++;
```

Comment:
```suggestion
            counter++;
            VerifyOrExit(counter < OPENTHREAD_CONFIG_MLE_IP_ADDRS_TO_REGISTER, OT_NOOP);
```
is it necessary ?

Reply:
I don't think it is necessary. 
`OPENTHREAD_CONFIG_MLE_IP_ADDRS_TO_REGISTER` equals to `OPENTHREAD_CONFIG_MLE_IP_ADDRS_PER_CHILD`, which is at least `4` according to Thread Conformance spec.
https://github.com/openthread/openthread/blob/779da7c858445d11df571dabfe344f11670c4109/src/core/config/mle.h#L87-L89

IIRC, there should be no test case requiring MTD to register less than maximal 4 entries in AR TLV, is there?